oprf/oprf.go

```diff
@@ -0,0 +1,55 @@
+package oprf
+
+import (
+	"github.com/cloudflare/circl/oprf/group"
+)
+
+// BlindToken corresponds to a token that has been blinded.
+// Internally, it is a serialized Point/Element.
+type BlindToken []byte
+
+// IssuedToken corresponds to a token that has been issued.
+// Internally, it is a serialized Point/Element.
+type IssuedToken []byte
+
+// Token is the object issuance of the protocol
+type Token struct {
+	data  []byte
+	blind *group.Scalar
+}
+
+// Evaluation corresponds to the evaluation over a token
+type Evaluation struct {
+	element []byte
+}
+
+// PublicKey is a struct representing a public key
+type PublicKey *group.Point
+
+// PrivateKey is a struct representing a private key
+type PrivateKey *group.Scalar
+
+// Client is
+type Client interface {
+	// SetUp configures a new client by generating its context and keys
+	SetUp(group.CiphersuiteID) (string, PrivateKey)
+
+	// Blind generates a token and blinded data.
+	Blind(in []byte) (*Token, *BlindToken)
+
+	// Unblind unblinds the server response.
+	Unblind(pk *PublicKey, t *Token, b *BlindToken, e *Evaluation) *IssuedToken
+}
+
+// Server is
+// TODO: should we add: VerifyFinalize and EvaluateBatch?
+type Server interface {
+	// SetUp configures a new server by generating its context and keys
+	SetUp(group.CiphersuiteID) (string, PrivateKey)
```

Comment:
As above, what are the output values?

Reply:
In Golang, the return values are defined after the method declaration, so following the definition, they are:

`SetUp(group.CiphersuiteID) (string, PrivateKey)`, meaning they are: a string and a PrivateKey. 

In Golang, there is no way to name the return values, unless you use this: https://yourbasic.org/golang/named-return-values-parameters/#:~:text=In%20Go%20return%20parameters%20may,are%20used%20as%20return%20values.&text=Named%20return%20parameters%20are%20initialized,can%20make%20for%20good%20documentation.

So, Im not really sure what you mean by what are them. As defined per spec, they are: a context and the key.  If you can, please explain me what you mean. Thanks ;)

Reply:
Copying the definition from the draft:

```
def SetupBaseServer(suite):
  (skS, _) = KeyGen(GG)
  contextString = I2OSP(modeBase, 1) || I2OSP(suite.ID, 2)
  return ServerContext(contextString, skS)
```

It returns a context, which is a string, and the private key.

Reply:
You can return a `Server` that implements `Evaluate` and `FullEvaluate`, the server structure is composed of two fields: the contextString and a skS.
For this to work, `SetupServer` must be a simple function with no receiver.